cpp/dolfin/mesh/cell_types.cpp

```diff
@@ -539,4 +538,20 @@ int mesh::cell_degree(mesh::CellType type, int num_nodes)
     throw std::runtime_error("Unknown cell type.");
   }
 }
+
+std::vector<int> mesh::cell_vertex_index(mesh::CellType type, int num_nodes,
+                                         int num_vertices_per_cell)
+{
+  int degree = mesh::cell_degree(type, num_nodes);
+  switch (type)
+  {
+  case mesh::CellType::quadrilateral:
```

Comment:
What about hexes?

Reply:
I wanted to add the Hex stuff separately. Its in https://github.com/FEniCS/dolfinx/commits/dokken/mesh-input-3D
I can of course add everything to this PR if that's whats easier for everyone.

Reply:
Presumably the hex case should be added here and an exception thrown?

Reply:
I'll just add the 3D stuff to this PR